client/main.lua

```diff
@@ -102,6 +102,64 @@ local function GetClosestCheckpoint()
     RaceData.ClosestCheckpoint = current
 end
 
+local function SetupPiles()
+    for k, v in pairs(CreatorData.Checkpoints) do
+        if CreatorData.Checkpoints[k].pileleft == nil then
+            ClearAreaOfObjects(v.offset.left.x, v.offset.left.y, v.offset.left.z, 50.0, 0)
+            CreatorData.Checkpoints[k].pileleft = CreateObject(`prop_offroad_tyres02`, v.offset.left.x, v.offset.left.y, v.offset.left.z, 0, 0, 0)
+            PlaceObjectOnGroundProperly(CreatorData.Checkpoints[k].pileleft)
+            FreezeEntityPosition(CreatorData.Checkpoints[k].pileleft, 1)
+            SetEntityAsMissionEntity(CreatorData.Checkpoints[k].pileleft, 1, 1)
+        end
+
+        if CreatorData.Checkpoints[k].pileright == nil then
+            ClearAreaOfObjects(v.offset.right.x, v.offset.right.y, v.offset.right.z, 50.0, 0)
+            CreatorData.Checkpoints[k].pileright = CreateObject(`prop_offroad_tyres02`, v.offset.right.x, v.offset.right.y, v.offset.right.z, 0, 0, 0)
+            PlaceObjectOnGroundProperly(CreatorData.Checkpoints[k].pileright)
+            FreezeEntityPosition(CreatorData.Checkpoints[k].pileleft, 1)
+            SetEntityAsMissionEntity(CreatorData.Checkpoints[k].pileleft, 1, 1)
+        end
+    end
+end
+
+local function CheckpointLoop()
+    CreateThread(function()
+        while RaceData.InCreator do
+            GetClosestCheckpoint()
+            SetupPiles()
+            Wait(1000)
+        end
+    end)
+end
+
+local function PylonsLoop()
+    CreateThread(function()
+        while RaceData.InCreator do
+            local PlayerPed = PlayerPedId()
+            local PlayerVeh = GetVehiclePedIsIn(PlayerPed, false)
+
+            if PlayerVeh ~= 0 then
+                local Offset = {
+                    left = {
+                        x = (GetOffsetFromEntityInWorldCoords(PlayerVeh, -CreatorData.TireDistance, 0.0, 0.0)).x,
+                        y = (GetOffsetFromEntityInWorldCoords(PlayerVeh, -CreatorData.TireDistance, 0.0, 0.0)).y,
+                        z = (GetOffsetFromEntityInWorldCoords(PlayerVeh, -CreatorData.TireDistance, 0.0, 0.0)).z,
+                    },
+                    right = {
+                        x = (GetOffsetFromEntityInWorldCoords(PlayerVeh, CreatorData.TireDistance, 0.0, 0.0)).x,
+                        y = (GetOffsetFromEntityInWorldCoords(PlayerVeh, CreatorData.TireDistance, 0.0, 0.0)).y,
+                        z = (GetOffsetFromEntityInWorldCoords(PlayerVeh, CreatorData.TireDistance, 0.0, 0.0)).z,
+                    }
+                }
+
+                DrawText3Ds(Offset.left.x, Offset.left.y, Offset.left.z, Lang:t('general.CheckL'))
+                DrawText3Ds(Offset.right.x, Offset.right.y, Offset.right.z, Lang:t('general.CheckR'))
+            end
+            Wait(3)
```

Comment:
Change to Wait(0), it has to run every frame, not every few

Reply:
Addressed in latest push